### PR TITLE
feat(alerts): Add time period selector to metric rule builder

### PR DESCRIPTION
**Pull request by scttcper**
Adds a time selector to the graph on the metric alert rule builder.

image

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsForm.tsx [hunk 1]
```diff
@@ -22,9 +23,7 @@ import {TimeWindow, IncidentRule, Dataset} from './types';
 import MetricField from './metricField';
 import {DATASET_EVENT_TYPE_FILTERS, DEFAULT_AGGREGATE} from './constants';
 
-type TimeWindowMapType = {[key in TimeWindow]: string};
-
-const TIME_WINDOW_MAP: TimeWindowMapType = {
+const TIME_WINDOW_MAP: Record<TimeWindow, string> = {
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsForm.tsx:26]: didn't know about this Record type in typescript, very nice

File: src/sentry/static/sentry/app/views/settings/incidentRules/triggers/chart/index.tsx [hunk 2]
```diff
@@ -95,44 +187,36 @@ class TriggersChart extends React.PureComponent<Props> {
...
   position: sticky;
   top: 69px; /* Height of settings breadcrumb 69px */
   z-index: ${p => p.theme.zIndex.dropdown - 1};
-  padding: ${space(2)} ${space(2)} 0;
-  border-bottom: 1px solid ${p => p.theme.borderLight};
   background: rgba(255, 255, 255, 0.9);
 `;
+
+const StyledPanel = styled(Panel)`
+  /* Can't have marign with the sticky window */
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/settings/incidentRules/triggers/chart/index.tsx:210]: nit: "margin"

**Review by davidenwang — APPROVED** [review]
lgtm

File: src/sentry/static/sentry/app/views/settings/incidentRules/triggers/chart/index.tsx (hunk 2 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/views/settings/incidentRules/triggers/chart/index.tsx:210]: marge
image

**Pull request merged**